llvm/lib/CodeGen/WindowScheduler.cpp

```diff
@@ -192,22 +192,36 @@ bool WindowScheduler::initialize() {
     return false;
   }
   // Check each MI in MBB.
-  SmallVector<Register, 8> PhiDefs;
+  SmallSet<Register, 8> PrevDefs;
+  SmallSet<Register, 8> PrevUses;
```

Comment:
These sets aren't used outside of the lambda. Just make the lambda a helper function with these used locally?

Reply:
These sets are used to record the defs and uses of phis that have already been checked in the current MBB.  
They need to be initialized and retain register information across multiple calls to the lambda. 
Therefore,  wouldn't it be better to place these sets outside the lambda?

Reply:
ok